crates/utils/src/logging.rs

```diff
@@ -1,29 +1,46 @@
+use std::str::FromStr;
+
 use anyhow::Result;
 use opentelemetry::trace::TracerProvider as _;
 use opentelemetry_otlp::WithTonicConfig;
 use opentelemetry_sdk::propagation::TraceContextPropagator;
-use tracing::subscriber::{self, Subscriber};
+use tracing::subscriber::Subscriber;
 use tracing_opentelemetry::OpenTelemetryLayer;
-use tracing_subscriber::{layer::SubscriberExt, EnvFilter, Layer, Registry};
+use tracing_subscriber::{
+    layer::{Filter, SubscriberExt},
+    Layer, Registry,
+};
+
+#[derive(Clone, Copy)]
+pub enum OpenTelemetry {
+    Enabled,
+    Disabled,
+}
+
+impl OpenTelemetry {
+    fn is_enabled(self) -> bool {
+        matches!(self, OpenTelemetry::Enabled)
+    }
+}
 
 /// Configures tracing and optionally enables an open-telemetry OTLP exporter.
```

Comment:
You can easily see this in the code (and the `env_or_default_filter` function) but this could also explain that there's an stdout layer and that filters are based on `RUST_LOG`.